tuf/download.py

```diff
@@ -35,29 +35,36 @@
 from __future__ import unicode_literals
 
 import os
-import socket
 import logging
 import time
 import timeit
-import ssl
 
-import tuf
+import requests
+import six
 
 import securesystemslib
 import securesystemslib.util
-import six
-
-# 'ssl.match_hostname' was added in Python 3.2.  The vendored version is needed
-# for Python 2.7.
-try:
-  from ssl import match_hostname
-
-except ImportError: # pragma: no cover
-  from securesystemslib._vendor.ssl_match_hostname import match_hostname
+import tuf
+import tuf.exceptions
 
 # See 'log.py' to learn how logging is handled in TUF.
 logger = logging.getLogger('tuf.download')
 
+# From http://docs.python-requests.org/en/master/user/advanced/#session-objects:
+#
+# "The Session object allows you to persist certain parameters across requests.
+# It also persists cookies across all requests made from the Session instance,
+# and will use urllib3's connection pooling. So if you're making several
+# requests to the same host, the underlying TCP connection will be reused,
+# which can result in a significant performance increase (see HTTP persistent
+# connection)."
+#
+# FIXME: There are very likely security implications to sharing the same
+# session. Ideally, tuf.client.updater would use a separate Session for every
+# mirror, and it would pass the Session to be used in this module, but that
+# requires a much more invasive change.
+#
+_session = requests.Session()
```

Comment:
Key to fix before merging.  Using a separate session for each hostname part of the URL should be easy.  Perhaps make this a dict?

Reply:
Each hostname might be a bit overkill --- I was thinking a separate session per repository mirror, but better to be safe than sorry. Let me work on this.